src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -0,0 +1,361 @@
+            }
+            logger.info(sampleName + ": Cutting (" + artifactMode.toString() +
+                    ") pre-preAdapterQ: " + String.valueOf(artifactModeNumToCut.get(artifactMode)));
+        }
+        return artifactModeNumToCut;
+    }
+
+    private static int calculateTotalNumToCut(final double fdrThresh, final long unfilteredGenotypeCount, final SortedMap<Genotype, VariantContext> genotypesToConsiderForFiltering) {
+        final List<Double> pArtifactScores = genotypesToConsiderForFiltering.keySet().stream()
+                .map(g -> OrientationBiasUtils.getGenotypeDouble(g, OrientationBiasFilterConstants.P_ARTIFACT_FIELD_NAME, 0.0))
+                .collect(Collectors.toList());
+
+        // When doing the Benjamini-Hochberg procedure, we need to include the non-artifact mode SNVs to guarantee the
+        //  FDR is kept to the specified threshold.
+        //   Pad the list of pArtifact scores so that the the non-artifacts (w/ a p-value of zero) are included.
+        final int numToPadZeroes = (int) unfilteredGenotypeCount - pArtifactScores.size();
+        List<Double> finalPArtifactScores = new ArrayList<>();
+        finalPArtifactScores.addAll(pArtifactScores);
+        finalPArtifactScores.addAll(Collections.nCopies(numToPadZeroes, 0.0));
+
+        return calculateTotalNumToCut(fdrThresh, unfilteredGenotypeCount, finalPArtifactScores);
+    }
+
+    /**
+     *
+     * @param fdrThresh desired maximum FDR threshold.  Must be greater than 0
+     * @param unfilteredGenotypeCount total number of unfiltered variants
+     * @param pArtifactScoresIncludingNonArtifact sorted list (descending) of the pArtifact scores.  Should include zeros for nonArtifact variants.
+     * @return total number of artifact mode variants to cut in order to be below the specified FDR threshold.
+     */
+    @VisibleForTesting
+    static int calculateTotalNumToCut(final double fdrThresh, final long unfilteredGenotypeCount, final List<Double> pArtifactScoresIncludingNonArtifact) {
+        ParamUtils.isPositive(fdrThresh, "FDR threshold must be positive and greater than zero.");
+
+        // Benjamini-Hochberg procedure https://en.wikipedia.org/wiki/False_discovery_rate#Benjamini.E2.80.93Hochberg_procedure
+        int result = pArtifactScoresIncludingNonArtifact.size() - 1;
+        for (int i = 0; i < pArtifactScoresIncludingNonArtifact.size(); i ++) {
+            final double pArtifact = pArtifactScoresIncludingNonArtifact.get(i);
+            if (pArtifact < (fdrThresh * (i + 1) / unfilteredGenotypeCount)) {
+                result = i;
+                break;
+            }
+        }
+        return result;
+    }
+
+    private static Map<ArtifactMode, Long> createArtifactModeCountMap(SortedSet<ArtifactMode> relevantArtifactModes, SortedMap<Genotype, VariantContext> genotypesToConsiderForFiltering) {
+        final Map<ArtifactMode, Long> artifactModeCount = new HashMap<>();
```

Comment:
Still not a fan of MutableInt due to limited performance boost, but not as versatile.  We can discuss...

No action.